### PR TITLE
S506-022 make the URIs package accessible to clients

### PR DESCRIPTION
**Pull request by setton**
This is very convenient for clients to be able to convert from
file names to URIs without having to reimplement their routines.

Also change the test for checking that file names are absolute to
one that works with final directory separators.

### CODE REVIEW

File: source/uri/uris.ads [hunk 1]
```diff
@@ -95,10 +95,7 @@ package URIs is
 
    package Conversions is
       function From_File (Full_Path : String) return URI_String
-        with Pre =>
-          Ada.Strings.Fixed.Equal_Case_Insensitive
-            (Ada.Directories.Full_Name (Full_Path),
-             Full_Path);
+        with Pre => GNAT.OS_Lib.Is_Absolute_Path (Full_Path);
```
Comment (godunko) [inline source/uri/uris.ads]: Good change but looks like unrelated to change "visibility" of URIs directory.

Comment (setton) [reply source/uri/uris.ads]: Right, but I did make a note of it in the commit changelog.

**Comment by setton** [comment]
> Also, did you checked that it doesn't break GPS build?

Yep!

File: source/uri/uris.ads (hunk 1 above)
Comment (reznikmm) [reply source/uri/uris.ads]: I'm not sure that implementation will still work with given precondition
